lib/puppet/provider/vcsrepo/git.rb

```diff
@@ -503,9 +504,16 @@ def set_excludes
     end
   end
 
+  # @!visibility private
+  def uncommitted_changes?
+    at_path do
+      git_with_identity('status', '--porcelain').empty?
```

Comment:
This change the sequence of commands run, you need to adjust CI to take this into account (see failed CI at the bottom of the "Conversation" tab).

---

Need guidance?  Read on :smiley: 

When reading the output from RSpec, you can identify 3 sections:
1. The output of the various tests with some details on failures;
2. A list of all failures details with backtraces;
3. A list of failed tests.

3 is the most handy to locate the tests you want to adjust (direct view of files and lines), while 1 & 2 tell you what is wrong.

If you consider [the first failure in the "third section"](https://github.com/puppetlabs/puppetlabs-vcsrepo/actions/runs/6228823069/job/17329274931?pr=623#step:6:1172), you see that the error occurred in this test:

https://github.com/puppetlabs/puppetlabs-vcsrepo/blob/2b05c3ab00253f3711f2329713c206234aa60875/spec/unit/puppet/provider/vcsrepo/git_spec.rb#L34-L44

If you scroll up [in the second section](https://github.com/puppetlabs/puppetlabs-vcsrepo/actions/runs/6228823069/job/17329274931?pr=623#step:6:915), you see the actual error:

```
  1) Puppet::Type::Vcsrepo::ProviderGit when with an ensure of present when with an ensure of present - with a revision that is a remote branch executes 'git clone' and 'git checkout -b'
     Failure/Error: exec_git(*args)

       Vcsrepo[test](provider=git) received :exec_git with unexpected arguments
         expected: ("branch", "--no-color", "-a")
              got: ("status", "--porcelain")
     # ./lib/puppet/provider/vcsrepo/git.rb:691:in `git_with_identity'
     # ./lib/puppet/provider/vcsrepo/git.rb:510:in `block in uncommitted_changes?'
     # ./lib/puppet/provider/vcsrepo.rb:53:in `block in at_path'
     # ./lib/puppet/provider/vcsrepo.rb:52:in `at_path'
     # ./lib/puppet/provider/vcsrepo/git.rb:509:in `uncommitted_changes?'
     # ./lib/puppet/provider/vcsrepo/git.rb:426:in `checkout'
     # ./lib/puppet/provider/vcsrepo/git.rb:22:in `create'
     # ./spec/unit/puppet/provider/vcsrepo/git_spec.rb:43:in `block (4 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.7.0/bin/rspec:23:in `load'
     # ./vendor/bundle/ruby/2.7.0/bin/rspec:23:in `<top (required)>'
     # /opt/hostedtoolcache/Ruby/2.7.8/x64/bin/bundle:23:in `load'
     # /opt/hostedtoolcache/Ruby/2.7.8/x64/bin/bundle:23:in `<main>'
```

In this case, adding a line like the following one between lines 40 and 41 should fix the issue:

```
   expect(provider).to receive(:exec_git).with('status', '--porcelain').and_return('') 
```

---

Thanks!